pkg/operator/mc.go

```diff
@@ -259,6 +259,36 @@ func (pc *ProfileCalculator) getPoolsForNode(node *corev1.Node) ([]*mcfgv1.Machi
 	return []*mcfgv1.MachineConfigPool{worker}, nil
 }
 
+// getNodesForPool returns a list of Nodes for MachineConfigPool 'pool'.
+func (pc *ProfileCalculator) getNodesForPool(pool *mcfgv1.MachineConfigPool) ([]*corev1.Node, error) {
+	selector, err := metav1.LabelSelectorAsSelector(pool.Spec.NodeSelector)
+	if err != nil {
+		return nil, fmt.Errorf("invalid label selector %s in MachineConfigPool %s: %v", util.ObjectInfo(selector), pool.ObjectMeta.Name, err)
+	}
+
+	initialNodes, err := pc.listers.Nodes.List(selector)
```

Comment:
I wonder if we should ignore nodes that are not ready or have scheduling disabled. Just to avoid a situation where a broken node (daemonset and/or kubelet) will prevent updates to the rest of the cluster.

Reply:
I do not think I agree here, this would unnecessarily complicate the design and I"m assuming that at least during the upgrades, it is all hands on deck and people should deal with broken nodes.

Reply:
But what if someone updates the PerfProfile while there is a node with stopped kubelet / nto daemonset? That Node object will have old bootcmdline data and will therefore prevent all the other nodes from updating to the new tuning.

Reply:
Yes, but we're talking about a normal operation (not upgrades).  I agree such nodes need to be excluded and I'll adjust this PR to reflect it.